### ISSUE TITLE
warning: unused parameter ‘sharedMemSize’

### ISSUE THREAD

**Issue by mxmlnkn** [ComputationalRadiationPhysics/cupla#38]
```C++
    cupla/include/cupla/kernel.hpp:72:17: warning: unused parameter ‘sharedMemSize’ [-Wunused-parameter]
             size_t sharedMemSize = 0,
                     ^
    cupla/include/cupla/kernel.hpp:82:10: warning: unused parameter ‘stream’ [-Wunused-parameter]
             cuplaStream_t stream = 0
              ^
```
This can be easily solved changing the corresponding lines to:
```C++
    static cuplaStream_t
    getStream(
        size_t /*sharedMemSize*/ = 0,
        cuplaStream_t stream = 0
    )
    {
        return stream;
    }

    static size_t
    getSharedMemSize(
        size_t sharedMemSize = 0,
        cuplaStream_t /*stream*/ = 0
    )
    {
        return sharedMemSize;
    }
```
The problem I have with these warnings, is that they get repeated for every single object file being compiled, because cupla is header only. This makes it hard to find real errors and useful warnings in user-code.

**Comment by ax3l** [comment]
sure, PR's are welcome!

**Comment by mxmlnkn** [comment]
fixed with #40
